mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -4627,10 +4633,19 @@ def _get_filter_clauses_for_search_traces(filter_string, session, dialect):
                 # which have key-value structure, so we need specialized handling
                 from mlflow.store.tracking.dbmodels.models import SqlSpan
 
-                span_column = getattr(SqlSpan, key_name)
-                val_filter = SearchTraceUtils.get_sql_comparison_func(comparator, dialect)(
-                    span_column, value
-                )
+                # Handle span.attributes.<attribute> format
+                if key_name.startswith("attributes."):
+                    attr_name = key_name[len("attributes.") :]
+                    # Search within the content JSON for the specific attribute
+                    # Using JSON extraction with LIKE for broad database compatibility
+                    val_filter = SearchTraceUtils.get_sql_comparison_func(comparator, dialect)(
+                        SqlSpan.content, f'%"mlflow.spanAttribute.{attr_name}"%{value}%'
```

Comment:
Does this work with `=` or `!=`? It seems the tests only check LIKE and ILIKE.

Reply:
It doesn't work, I updated the PR description.